src/main/java/seedu/address/ui/CommandHistory.java

```diff
@@ -52,7 +57,11 @@ public String getNext() {
         if (hasNext()) {
             return history.get(moveForward());
         } else {
+            if (hasRudimentary()) {
+                return getRudimentary();
+            }
             throw new NoSuchElementException();
+
         }
```

Comment:
would be clearer as a else if (hasRudimentary())  + throwing outside of the if/elseif block